### ISSUE TITLE
Code Debt: Check `make swagger` in CI

### ISSUE THREAD

**Issue by Pinjasaur** [mattermost/focalboard#4190]
Similar to #3096, check `make swagger` in CI to verify the API docs stay up to date. The embedded Swagger YAML definitions inside Golang comments can be a bit fragile with vague error messages being hard to follow.

Rough CI steps:

1. `make swgger` (this may error on it's own)
2. Check if changes e.g. `git status --porcelain` or `git diff --exit-code`

**Comment by Pinjasaur** [comment]
It would also be worth evaluating if maybe we want to split out the Swagger command(s) into 2 `make` targets. Since currently `make swagger` generates the `swagger.yml` _plus_ all the "clients" e.g. language clients and HTML documentation. The `server/swagger/clients` directory is gitignored so all the language clients currently get generated for naught—except to serve as a smoketest.

**Comment by mastersb** [comment]
While working on #4151 
I ran `make swagger` to generate the API docs, and when it reached the point of generating the Python client with:
`cd server/swagger && openapi-generator generate -i swagger.yml -g python -o clients/python`
I encountered the following Java exception

<details>
  <summary>Click to expand</summary>


```
Exception in thread "main" java.lang.RuntimeException: Could not generate api file for 'default'
	at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:695)
	at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:915)
	at org.openapitools.codegen.cmd.Generate.execute(Generate.java:465)
	at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
	at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
Caused by: com.github.jknack.handlebars.HandlebarsException: endpoint.handlebars:3:3: java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.util.HashMap$Node[] java.util.HashMap.table accessible: module java.base does not "opens java.util" to unnamed module @54307ae0
    endpoint.handlebars:3:3
	at java.base/java.lang.reflect.AccessibleObject.throwInaccessibleObjectException(AccessibleObject.java:387)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:363)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:311)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:180)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:174)
	at com.github.jknack.handlebars.context.FieldValueResolver$FieldMember.setAccessible(FieldValueResolver.java:150)
	at com.github.jknack.handlebars.context.MemberValueResolver.cache(MemberValueResolver.java:82)
	at com.github.jknack.handlebars.context.MemberValueResolver.resolve(MemberValueResolver.java:54)
	at com.github.jknack.handlebars.Context$CompositeValueResolver.resolve(Context.java:199)
	at com.github.jknack.handlebars.internal.path.PropertyPath.eval(PropertyPath.java:52)
	at com.github.jknack.handlebars.Context$PathExpressionChain.next(Context.java:361)
	at com.github.jknack.handlebars.Context$PathExpressionChain.eval(Context.java:381)
	at com.github.jknack.handlebars.Context.get(Context.java:618)
	at com.github.jknack.handlebars.internal.Partial.override(Partial.java:253)
	at com.github.jknack.handlebars.internal.Partial.merge(Partial.java:226)
	at com.github.jknack.handlebars.internal.BaseTemplate.apply(BaseTemplate.java:125)
	at com.github.jknack.handlebars.internal.TemplateList.merge(TemplateList.java:95)
	at com.github.jknack.handlebars.internal.BaseTemplate.apply(BaseTemplate.java:125)
	at com.github.jknack.handlebars.internal.BaseTemplate.apply(BaseTemplate.java:113)
	at org.openapitools.codegen.templating.HandlebarsEngineAdapter.compileTemplate(HandlebarsEngineAdapter.java:92)
	at org.openapitools.codegen.TemplateManager.write(TemplateManager.java:163)
	at org.openapitools.codegen.languages.PythonClientCodegen.processTemplateToFile(PythonClientCodegen.java:480)
	at org.openapitools.codegen.languages.PythonClientCodegen.generateFiles(PythonClientCodegen.java:503)
	at org.openapitools.codegen.languages.PythonClientCodegen.generateEndpoints(PythonClientCodegen.java:669)
	at org.openapitools.codegen.languages.PythonClientCodegen.postProcessOperationsWithModels(PythonClientCodegen.java:914)
	at org.openapitools.codegen.DefaultGenerator.processOperations(DefaultGenerator.java:1237)
	at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:599)
	... 4 more
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.util.HashMap$Node[] java.util.HashMap.table accessible: module java.base does not "opens java.util" to unnamed module @54307ae0
	... 31 more
make: *** [swagger] Error 1
```
</details>

I'm on macOS 12.6.1 and installed go-swagger and openapi-generator via the instructions [here](https://github.com/mattermost/focalboard/blob/main/server/swagger/README.md)
go-swagger: stable 0.30.3
openapi-generator: stable 6.2.1

Happy to provide any additional info to aid in troubleshooting this.

**Comment by Pinjasaur** [comment]
I updated my mine locally with a `brew update` and `brew upgrade` and am seeing the same behavior too. Doing a bit of digging isn't revealing any obvious solutions that I'm seeing.

I'm wondering if it's possible it's a problem on the OpenAPI / Swagger side with that specific generator for Python. Since the other clients seem to be generating OK (or at least without errors).

**Comment by Pinjasaur** [comment]
This appears to the closest related thing I've been able to find. Not positive it's the same issue, though. https://github.com/OpenAPITools/openapi-generator/issues/11763
